formats/config/src/main/kotlin/kotlinx/serialization/config/ConfigParser.kt

```diff
@@ -21,7 +21,9 @@ public class ConfigParser(
     public inline fun <reified T : Any> parse(conf: Config): T = parse(conf, context.getContextualOrDefault(T::class))
 
     public fun <T> parse(conf: Config, deserializer: DeserializationStrategy<T>): T =
-        ConfigReader(conf).decode(deserializer)
+        ConfigDecoder(conf).decode(deserializer)
+
+    public interface ConfigInput : Decoder, CompositeDecoder {}
```

Comment:
Am I right that you need this interface to perform format-specific operations in a same manner as with `JsonInput`? Can you please tell me more about your use-case? This interface does not have any useful methods that can be used during `deserialize()`. 

Also, pay attention that it is recommended to have documentation for every public interface.